wp_api_integration_tests/src/lib.rs

```diff
@@ -165,3 +165,17 @@ impl<T: std::fmt::Debug, E: std::error::Error> AssertResponse for Result<T, E> {
         self.unwrap()
     }
 }
+
+pub trait AssertError {
+    type Item;
+    fn assert_error(self) -> Self::Item;
+}
+
+impl<T: std::fmt::Debug, E: std::error::Error> AssertError for Result<T, E> {
+    type Item = E;
+
+    fn assert_error(self) -> E {
+        assert!(self.is_err(), "Request was successful");
+        self.unwrap_err()
+    }
+}
```

Comment:
This is probably not going to be common enough to have a `trait` extension. Also, the `Request was successful` might throw us of in the test logs, because it's not clear that it was expected to fail.

In most cases, we need to validate that the request failed with a certain error for which we already have `AssertWpError` trait. So, at least for now, I think we should individually implement this assertion and pass in a message that'll point out the issue. If there is enough common usage, then we can implement a specialized one like `AssertWpError`.

I realize that we discussed me taking over these PRs and implement them as I see fit, but I just wanted to drop my thoughts here, so if I end up changing it, it doesn't come out of nowhere. 😅